frontend/src/Components/UserProfileSettings.tsx

```diff
@@ -51,17 +54,81 @@ export default function UserProfileSettings(props: UserProfileSettingsProps) {
     });
   };
 
-  return (
-    <div>
-      {gender !== undefined && <button className={buttonStyles.logoutButton} onClick={handleGenderChange}><UserIcon /> Пол: {
-        gender === UserGender.fluid ? 'не указан' : (
-          gender === UserGender.she ? 'женщина' : (
-            'мужчина'
-          )
-        )
-      } </button>}
-      {<ThemeToggleComponent dynamic={true} buttonLabel='Сменить тему' /> }
-      {<button className={buttonStyles.logoutButton} onClick={handleLogout}><LogoutIcon /> Выйти </button> }
-    </div>
-  );
+    return (
+        <>
+            <div>
+                {gender !== undefined &&
+                    <button className={buttonStyles.logoutButton} onClick={handleGenderChange}><UserIcon/> Пол: {
+                        gender === UserGender.fluid ? 'не указан' : (
+                            gender === UserGender.she ? 'женщина' : (
+                                'мужчина'
+                            )
+                        )
+                    } </button>}
+                {<ThemeToggleComponent dynamic={true} buttonLabel="Сменить тему"/>}
+                {<button className={buttonStyles.logoutButton} onClick={handleLogout}><LogoutIcon/> Выйти </button>}
+            </div>
+            {props.barmaliniAccess && <BarmaliniAccess/>}
+        </>
+    );
 }
+
+
+/**
+ * Component that displays the button, when clicked,
+ * the button turns into a div with the password/token (got from the server via userAPi.getBarmaliniPassword)
+ * and the "copy" button.
+ */
+const BarmaliniAccess = observer(() => {
+    const api = useAPI();
+    const [access, setAccess] = React.useState<BarmaliniAccessResult | undefined>();
+
+
+    const handleCopy = (e: React.MouseEvent) => {
+        e.preventDefault();
+        if (access === undefined) {
+            return;
+        }
+        navigator.clipboard?.writeText(access.password)
+            ?.then(() => toast('В буфере!'))?.catch();
+    };
+
+    const handleShowPassword = (e: React.MouseEvent) => {
+        e.preventDefault();
+        if (!access) {
+            api.userAPI.getBarmaliniAccess().then((data) => {
+                setAccess(data);
+            }).catch((error) => {
+                toast.error(error?.message || 'Не удалось получить пароль.');
+            });
+        } else {
+            setAccess(undefined);
+        }
+    };
+
+    // selects all text in the element
+    const selectText = (e: React.MouseEvent) => {
+        e.preventDefault();
+        const element = e.target as HTMLElement;
+        const range = document.createRange();
+        range.selectNodeContents(element);
+        const selection = window.getSelection();
+        if (selection) {
+            selection.removeAllRanges();
+            selection.addRange(range);
+        }
+    };
+
+    return (
+        <div className={styles.barmalini}>
+            {access && <div>
+                <div><span className={styles.label}>Логин:</span> {access.login}</div>
```

Comment:
It is worth noting that password is valid for only one hour.
```suggestion
                <div>
                Счастливого бармаления. Пароль истекает через час.<br />
                <span className={styles.label}>Логин:</span> {access.login}</div>
```